ISS-257: Add define_variable_has_comment attribute to support comment existence check

nhaydel opened the pull request:
Steps to test:

Run unit tests

nhaydel commented:
@gerrycampion since this Issue can be solved multiple ways:
1. Using the expected_variables operation in conjunction with the new define_variable_has_comment attribute
2. Using the result of merging define variable metadata with library variable metadata

And I believe we agreed that we prefer option 2 since it is more robust. I will take this time to implement the new rule type for define variable metadata with library variable metadata. If you agree with this approach I will convert this PR to draft until I have implemented the additional rule type

gerrycampion commented:
> @gerrycampion since this Issue can be solved multiple ways:
> 
> 1. Using the expected_variables operation in conjunction with the new define_variable_has_comment attribute
> 2. Using the result of merging define variable metadata with library variable metadata
> 
> And I believe we agreed that we prefer option 2 since it is more robust. I will take this time to implement the new rule type for define variable metadata with library variable metadata. If you agree with this approach I will convert this PR to draft until I have implemented the additional rule type

Yes, agreed

nhaydel commented:
> > @gerrycampion since this Issue can be solved multiple ways:
> > 
> > 1. Using the expected_variables operation in conjunction with the new define_variable_has_comment attribute
> > 2. Using the result of merging define variable metadata with library variable metadata
> > 
> > And I believe we agreed that we prefer option 2 since it is more robust. I will take this time to implement the new rule type for define variable metadata with library variable metadata. If you agree with this approach I will convert this PR to draft until I have implemented the additional rule type
> 
> Yes, agreed

Added new rule type: `Define Item Metadata Check against Library Metadata`

gerrycampion commented on cdisc_rules_engine/dataset_builders/define_variables_with_library_metadata.py:
```diff
@@ -0,0 +1,46 @@
+        "define_variable_order_number",
+        "define_variable_has_codelist",
+        "define_variable_codelist_coded_values",
+        "define_variable_mandatory",
```
add "define_variable_has_comment",

gerrycampion commented on resources/schema/MetaVariables.md:
```diff
@@ -66,6 +66,10 @@ ItemGroupDef.ItemDef.DataType
 
 ItemGroupDef.ItemDef.CodeListRef exists
 
+## define_variable_has_comment
```
Need to also create schema and documentation for `define_vlm_has_comment`

gerrycampion commented on resources/schema/MetaVariables.md:
```diff
@@ -130,6 +134,10 @@ ValueListDef.ItemDef.DataType
 
 ValueListDef.ItemDef.CodeListRef exists
 
+## define_variable_has_comment
```
should be define_vlm_has_comment

gerrycampion commented:
For QA, I tried testing with the attached rule and data. I expect LB to return a issue for LBCAT, but instead I am receiving an error:
```json
[
  {
    "executionStatus": "execution_error",
    "domain": "LB",
    "variables": [],
    "message": "rule execution error",
    "errors": [
      {
        "error": "Column not found in data",
        "message": "ordinal"
      }
    ]
  }
]
```
[Rule (3).zip](https://github.com/cdisc-org/cdisc-rules-engine/files/12457691/Rule.3.zip)

The same data is also available in the prod rule editor and sharepoint under SD1149

nhaydel commented:
@gerrycampion with the latest update we are no longer seeing this issue. I believe it was an issue with how we were passing library metadata to the rule builders
